### PR TITLE
NIFI-10842 Adding option to disable HTTP2 in StandardOauth2AccessTokenProvider

### PR DESCRIPTION
**Pull request by ruckc**
# Summary

[NIFI-10842](https://issues.apache.org/jira/browse/NIFI-10842)

# Tracking

Please complete the following tracking steps prior to pull request creation.

### Issue Tracking

- [X] [Apache NiFi Jira](https://issues.apache.org/jira/browse/NIFI) issue created

### Pull Request Tracking

- [X] Pull Request title starts with Apache NiFi Jira issue number, such as `NIFI-00000`
- [X] Pull Request commit message starts with Apache NiFi Jira issue number, as such `NIFI-00000`

### Pull Request Formatting

- [X] Pull Request based on current revision of the `main` branch
- [X] Pull Request refers to a feature branch with one commit containing changes

# Verification

Please indicate the verification steps performed prior to pull request creation.

### Build

- [X] Build completed using `mvn clean install -P contrib-check`
  - [X] JDK 8
  - [X] JDK 11
  - [X] JDK 17

### Licensing

- [X] New dependencies are compatible with the [Apache License 2.0](https://apache.org/licenses/LICENSE-2.0) according to the [License Policy](https://www.apache.org/legal/resolved.html)
- [X] New dependencies are documented in applicable `LICENSE` and `NOTICE` files

### Documentation

- [X] Documentation formatting appears as expected in rendered files

### CODE REVIEW

**Comment by MikeThomsen** [comment]
@ruckc FYI

image

File: nifi-nar-bundles/nifi-standard-services/nifi-oauth2-provider-bundle/nifi-oauth2-provider-service/src/main/java/org/apache/nifi/oauth2/HttpProtocolStrategy.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.oauth2;
+
+import org.apache.nifi.components.DescribedValue;
+import org.apache.nifi.jetty.configuration.connector.ApplicationLayerProtocol;
```
Comment (exceptionfactory) [inline nifi-nar-bundles/nifi-standard-services/nifi-oauth2-provider-bundle/nifi-oauth2-provider-service/src/main/java/org/apache/nifi/oauth2/HttpProtocolStrategy.java:20]: It looks like this import needs to be removed

**Review by exceptionfactory — CHANGES_REQUESTED** [review]

File: nifi-nar-bundles/nifi-standard-services/nifi-oauth2-provider-bundle/nifi-oauth2-provider-service/src/main/java/org/apache/nifi/oauth2/HttpProtocolStrategy.java (hunk 1 above)
Comment (ruckc) [reply nifi-nar-bundles/nifi-standard-services/nifi-oauth2-provider-bundle/nifi-oauth2-provider-service/src/main/java/org/apache/nifi/oauth2/HttpProtocolStrategy.java:20]: yep, working on it.  was just getting the PR started while I sort out all the details locally.

**Review by exceptionfactory — CHANGES_REQUESTED** [review]
Thanks for making the adjustments @ruckc, this looks close to completion.

I noted one adjustment for the default property value. The current OkHttp client supports both HTTP/2 and HTTP/1.1, so that should be the default value for the new property.

File: nifi-nar-bundles/nifi-standard-services/nifi-oauth2-provider-bundle/nifi-oauth2-provider-service/src/main/java/org/apache/nifi/oauth2/StandardOauth2AccessTokenProvider.java [hunk 2]
```diff
@@ -151,8 +152,18 @@ public class StandardOauth2AccessTokenProvider extends AbstractControllerService
         .required(false)
         .build();
 
+    public static final PropertyDescriptor HTTP_PROTOCOL_STRATEGY = new PropertyDescriptor.Builder()
+        .name("HTTP Protocols")
+        .description("HTTP Protocols supported for Application Layer Protocol Negotiation with TLS")
+        .required(true)
+        .allowableValues(HttpProtocolStrategy.class)
+        .defaultValue(HttpProtocolStrategy.HTTP_1_1.getValue())
```
[the remark this answers is not in the document]
Comment (ruckc) [reply nifi-nar-bundles/nifi-standard-services/nifi-oauth2-provider-bundle/nifi-oauth2-provider-service/src/main/java/org/apache/nifi/oauth2/StandardOauth2AccessTokenProvider.java:160]: Good catch, just fixed it.

**Review by exceptionfactory — APPROVED** [review]
Thanks for your work on this @ruckc! The latest version still appears to reference `HTTP_1_1` as the default value instead of `H2_HTTP_1_1`, but I will make that change when merging. +1

**Pull request closed without merging**